awpy/visualization/plot.py

```diff
@@ -326,3 +341,382 @@ def plot_nades(
+
+        # Save current frame map control viz to file
+        # All frames are saved to './tmp/ folder '
+        plot_frame_map_control(
+            map_name,
+            frame,
+            plot_type=plot_type,
+            save_filepath=filename,
+        )
+
+        # Load image back as frame of gif that will
+        # be created at the end of this function
+        images.append(imageio.imread(filename))
+        i += 1
+
+    print("Creating gif!")
+    imageio.imwrite(gif_filepath, images)
+
+
+def _plot_map_control_metrics_helper(
+    metrics: list[float],
+    axes: plt.Axes,
+) -> None:
+    """Helper function to plot map control metrics.
+
+    Args:
+        metrics (list): List containing map control values to plot
+        axes (axes): axes object for plotting
+
+    Returns: Nothing, all plotting is done on ax_object
+
+    Raises:
+        ValueError: If metrics is empty
+    """
+    if not metrics:
+        msg = "Metrics is empty."
+        raise ValueError(msg)
+
+    x = list(range(1, len(metrics) + 1))
+    axes.plot(x, metrics)
+    axes.set_ylim(0, 1)
+    axes.axhline(y=0.5, linestyle="--", c="k")
+    axes.set_ylabel("Map Control Metric Value")
+    axes.set_xlabel("Frame Number")
+    axes.set_title("Map Control Metric Progress")
+    x_tick_threshold = 10
+    if len(metrics) > x_tick_threshold:
+        step_size = int(len(metrics) // x_tick_threshold)
+        axes.set_xticks(list(range(1, len(metrics) + 1, step_size)))
+
+
+def plot_map_control_metrics(
+    metric_arr: list[float],
+    given_fig_ax: tuple[plt.Figure, plt.Axes] | tuple[None, None] = (None, None),
+) -> None:
+    """Function to plot given map control metrics.
+
+    Args:
+        metric_arr (list): List containing map control values to plot
+        given_fig_ax (tuple): Fig and ax objects if given
+
+    Returns: Plot given map control metric values onto axes object
+
+    Raises:
+        ValueError: If metrics is empty
+    """
+    if not metric_arr:
+        msg = "Metrics is empty."
+        raise ValueError(msg)
+
+    if given_fig_ax[0] is not None:
+        _plot_map_control_metrics_helper(metric_arr, given_fig_ax[1])
+    else:
+        _, curr_ax = plt.subplots()
+        _plot_map_control_metrics_helper(metric_arr, curr_ax)
+        plt.show()
+
+
+def _save_map_control_graphic_helper(
+    map_name: str,
+    frame: GameFrame,
+    metrics: list[float],
+    save_path: str = "",
+) -> None:
+    """Helper function to generate map control graphic for awpy round.
+
+    Args:
+        map_name (str): Map used in frame_map_control_metric call
+        frame (GameFrame): awpy frame to create map control graphic
+        metrics  (list): List containing map control values to plot
+        save_path (str): Location where graphic should be saved on file
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
```

Comment:
Should only need to happen at the public interface.

Reply:
I think I addressed this already in a commit on the branch.